install.sh

```diff
@@ -507,41 +509,15 @@ fi
 # shellcheck disable=SC2016
 ohai 'Checking for `sudo` access (which may request your password)...'
 
-if [[ -z "${HOMEBREW_ON_LINUX-}" ]]
+if [[ -n "${HOMEBREW_ON_MACOS-}" ]]
 then
   have_sudo_access
-else
-  if [[ -w "${HOMEBREW_PREFIX_DEFAULT}" ]] ||
-     [[ -w "/home/linuxbrew" ]] ||
-     [[ -w "/home" ]]
-  then
-    HOMEBREW_PREFIX="${HOMEBREW_PREFIX_DEFAULT}"
-  elif [[ -n "${NONINTERACTIVE-}" ]]
-  then
-    if have_sudo_access
-    then
-      HOMEBREW_PREFIX="${HOMEBREW_PREFIX_DEFAULT}"
-    else
-      abort "Insufficient permissions to install Homebrew to \"${HOMEBREW_PREFIX_DEFAULT}\"."
-    fi
-  else
-    trap exit SIGINT
-    if ! /usr/bin/sudo -n -v &>/dev/null
-    then
-      ohai "Select a Homebrew installation directory:"
-      echo "- ${tty_bold}Enter your password${tty_reset} to install to ${tty_underline}${HOMEBREW_PREFIX_DEFAULT}${tty_reset} (${tty_bold}recommended${tty_reset})"
-      echo "- ${tty_bold}Press Control-D${tty_reset} to install to ${tty_underline}${HOME}/.linuxbrew${tty_reset}"
-      echo "- ${tty_bold}Press Control-C${tty_reset} to cancel installation"
-    fi
-    if have_sudo_access
-    then
-      HOMEBREW_PREFIX="${HOMEBREW_PREFIX_DEFAULT}"
-    else
-      HOMEBREW_PREFIX="${HOME}/.linuxbrew"
-    fi
-    trap - SIGINT
-  fi
-  HOMEBREW_REPOSITORY="${HOMEBREW_PREFIX}/Homebrew"
+elif ! [[ -w "${HOMEBREW_PREFIX}" ]] &&
+     ! [[ -w "/home/linuxbrew" ]] &&
+     ! [[ -w "/home" ]] &&
+     ! have_sudo_access
+then
+  abort "Insufficient permissions to install Homebrew to \"${HOMEBREW_PREFIX}\"."
```

Comment:
> Let's not encourage it.

Should we add a link <https://docs.brew.sh/Installation#alternative-installs> here to tell people how to install Homebrew without `sudo` privilege? One of the reasons the user using Homebrew is they don't have `sudo` privilege.

Reply:
> It'll be clearer that this is unsupported after Homebrew/brew#13971

I see, we only support default prefixes both on macOS and Linux now. If non-default prefixes are not officially supported, does that means we will not ship relocatable bottles in the future or the priority is lowered?

Reply:
@XuehaiPan We've only ever supported default prefix/bottles on Linux, we've just been bad at wording it.

If we can get to the point where we have roughly >=90% bottles relocatable outside the default prefix: we'll reconsider what we support.

Our support is pretty much based on "on what configurations can we continue to provide bottles".